python/metatensor-operations/metatensor/operations/solve.py

```diff
@@ -93,7 +93,7 @@ def _solve_block(X: TensorBlock, Y: TensorBlock) -> TensorBlock:
     """
     # TODO handle properties and samples not in the same order?
 
-    if not np.all(X.samples == Y.samples):
+    if not _dispatch.all(X.samples.values == Y.samples.values):
```

Comment:
```suggestion
    if not X.samples == Y.samples:
```

this was an old leftover, we can now compare Labels directly